lib/centurion/deploy_callbacks.rb

```diff
@@ -0,0 +1,41 @@
+module Centurion
+  # Callbacks to allow hooking into the deploy lifecycle. This could
+  # be useful to communicate with a loadbalancer, chat room, etc.
+  module DeployCallbacks
+    def stop_containers(server, service, timeout = 30)
+      emit :before_stopping_container, server, service
+      super server, service, timeout
+    end
+
+    def before_starting_container(server, service)
+      emit :before_starting_container, server, service
+    end
+
+    def start_new_container(server, service, restart_policy)
+      super(server, service, restart_policy).tap { emit :after_starting_container, server, service }
+    end
+
+    def wait_for_health_check_ok(health_check_method, server, port, endpoint, image_id, tag, sleep_time=5, retries=12)
+      super(health_check_method,
+            server,
+            port,
+            endpoint,
+            image_id,
+            tag,
+            sleep_time,
+            retries).tap { emit :after_health_check_ok, server }
+    end
+
+    private
+
+    def emit(name, *args)
+      callbacks[name].each do |callback|
+        callback.call(*args)
+      end
+    end
+
+    def callbacks
+      fetch 'callbacks', Hash.new { [] }
```

Comment:
Well the idea had been to keep `fetch` and other env methods out of that code entirely and to have the required data passed in instead. I'd need to spend some more time looking at this code to see what the right thing is to do there.